### ISSUE TITLE
Performance issue with datapath.

### ISSUE THREAD

**Issue by oqu** [Kmotiko/gofc#11]
https://github.com/Kmotiko/gofc/blob/e7cf943c3c3a932c815332e9996da11ed24de13d/datapath.go#L44-L45
I believe the "default: continue" should be removed. If there is nothing to send, we are going to 'waste' time in this default case.

**Comment by Kmotiko** [comment]
Thank you for reporting a issue.
I still haven't measure performance yet , but I think it should be removed as you say.
I'll confirm the behavior and fix sendLoop function.

Incidentally, in future, sendLoop function should wait shutdown-notification to do graceful-shutdown like this.

```
select {
	// wait channel
	case msg := <-(dp.sendBuffer):
		// handle message
		...
	// wait shutdown notification
	case <- dp.shutdown:
		// handle shutdown
}
```

**Comment by Kmotiko** [comment]
@oqu 
I remove default case from sendLoop and create PullRequest as #13.
I would be helpful if you confirm my changing.
If there are no problem, I'll merge that.

I treat graceful-shutdown feature  as future work.

**Comment by Kmotiko** [comment]
Is there any update on this ?

**Comment by Kmotiko** [comment]
Sorry, I miss your comment in pull request thread ...
I'll close this issue after merge #13

**Comment by Kmotiko** [comment]
I merged #13, so close this issue.
